### PR TITLE
PIN 3496 Api-Gateway Migration

### PR DESCRIPTION
**Pull request by nttdata-rtorsoli**

### CODE REVIEW

File: src/main/scala/it/pagopa/interop/apigateway/api/impl/GatewayApiServiceImpl.scala [hunk 1]
```diff
@@ -115,27 +113,31 @@ final case class GatewayApiServiceImpl(
...
+    import AgreementProcessDependency.AgreementState._
     val operationLabel =
       s"Retrieving agreements for producerId $producerId consumerId $consumerId eServiceId $eServiceId descriptorId $descriptorId states $states"
     logger.info(operationLabel)
 
     val result: Future[Agreements] = for {
+      producerUuid    <- producerId.traverse(_.toFutureUUID)
+      consumerUuid    <- consumerId.traverse(_.toFutureUUID)
+      eServiceUuid    <- eServiceId.traverse(_.toFutureUUID)
+      descriptorUuid  <- descriptorId.traverse(_.toFutureUUID)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/apigateway/api/impl/GatewayApiServiceImpl.scala]: ⚠️ cats helper with Future

File: src/main/scala/it/pagopa/interop/apigateway/api/impl/package.scala [hunk 2]
```diff
@@ -88,13 +82,15 @@ package object impl extends SprayJsonSupport with DefaultJsonProtocol {
...
-  implicit class EnrichedPurpose(private val purpose: PurposeManagementApiPurpose) extends AnyVal {
-    def toModel: Try[Purpose] =
+  implicit class EnrichedPurpose(private val purpose: PurposeProcessApiPurpose) extends AnyVal {
+    def toModel: Either[Throwable, Purpose] = {
       purpose.versions
         .sortBy(_.createdAt)
         .find(p => p.state == PurposeVersionState.ACTIVE || p.state == PurposeVersionState.SUSPENDED)
-        .toTry(MissingActivePurposeVersion(purpose.id))
-        .map(version => Purpose(id = purpose.id, throughput = version.dailyCalls, state = version.state.toModel))
+        .fold(Left(MissingActivePurposeVersion(purpose.id)): Either[Throwable, Purpose]) { v =>
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/apigateway/api/impl/package.scala]:
```suggestion
        .fold[Either[Throwable, Purpose]](Left(MissingActivePurposeVersion(purpose.id))) { v =>
```

File: src/main/scala/it/pagopa/interop/apigateway/service/AttributeRegistryProcessService.scala [hunk 3]
```diff
@@ -1,14 +1,31 @@
...
+    def getAttributesFrom(offset: Int): Future[Seq[Attribute]] =
+      getBulkAttributes(attributeIds = attributeIds, offset = offset, limit = 50).map(_.results)
+
+    def go(start: Int)(as: Seq[Attribute]): Future[Seq[Attribute]] =
+      getAttributesFrom(start).flatMap(attrs =>
+        if (attrs.size < 50) Future.successful(as ++ attrs) else go(start + 50)(as ++ attrs)
+      )
+
+    go(0)(Nil)
+  }
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/apigateway/service/AttributeRegistryProcessService.scala]: The implementation should be (and is already) in the `Impl` class
```suggestion
```

File: src/main/scala/it/pagopa/interop/apigateway/api/impl/GatewayApiServiceImpl.scala [hunk 4]
```diff
@@ -505,7 +505,7 @@ final case class GatewayApiServiceImpl(
       latestDescriptor <- eService.latestAvailableDescriptor
       state            <- latestDescriptor.state.toModel.toFuture
       allAttributesIds = latestDescriptor.attributes.allIds
-      attributes <- attributeRegistryProcessService.getBulkAttributes(allAttributesIds)
+      attributes <- attributeRegistryProcessService.loadBulkAttributes(allAttributesIds)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/apigateway/api/impl/GatewayApiServiceImpl.scala]:
```suggestion
      attributes <- attributeRegistryProcessService.getBulkAttributes(allAttributesIds)
```
Note: load is the logic that creates new attributes starting from IPA

**Review by galales — CHANGES_REQUESTED** [review]

**Review by beetlecrunch — APPROVED** [review]
I approve, but the changes indicated by @galales are necessary

**Review by TonioGela — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/apigateway/service/AttributeRegistryProcessService.scala (hunk 3 above)
Comment (nttdata-rtorsoli) [reply src/main/scala/it/pagopa/interop/apigateway/service/AttributeRegistryProcessService.scala]: I didn't understand

**Review by galales — APPROVED** [review]

**Pull request merged**